packages/gnome-shell/src/extensions/sharedInternals.d.ts

```diff
@@ -4,6 +4,12 @@ import type { Extension } from './extension.js';
 import type Gio from '@girs/gio-2.0';
 import type GObject from '@girs/gobject-2.0';
 
+/**
+ * @see https://gjs.guide/extensions/development/translations.html#marking-strings-for-translation
+ * @notes in Gnome Shell every string has format, since String.prototype.format() is added by it, but the best practices state, that only gettext functions use that, we already have that types in  "global.d.ts" but if you don't want to have that type on every string, it is the return type of all those functions without importing the global types 
+ */
+export type FormattableString = string & { format: (...args: any[]) => string };
```

Comment:
Personally I don't like this change.  Type declarations should reflect how the API actually is, not how it's believed to be intended.

Gnome shell monkey patches string, and the types should reflect that, because it could leak into places beyond just gettext.

If you wish to force tighter restrictions to maintain a certain paradigm in your own code, you can always ship your own declaration files

Reply:
Alright, so I remove that, 👍🏼

Reply:
Perhaps propose it in a separate pull request? It's just my opinion after all, perhaps other like it, and I think it's worth discussing.👍

But let's keep it separate from the GNOME 46 updates which don't need any discussion 🙂